watertap/property_models/anaerobic_digestion/adm1_reactions.py

```diff
@@ -0,0 +1,1433 @@
+Department of Industrial Electrical Engineering and Automation, Lund University, Lund, Sweden. (2006) 1–35.
+
+"""
+
+# Import Pyomo libraries
+import pyomo.environ as pyo
+
+# Import IDAES cores
+from idaes.core import (
+    declare_process_block_class,
+    MaterialFlowBasis,
+    ReactionParameterBlock,
+    ReactionBlockDataBase,
+    ReactionBlockBase,
+)
+from idaes.core.util.misc import add_object_reference
+from idaes.core.util.exceptions import BurntToast
+import idaes.logger as idaeslog
+
+
+# Some more information about this module
+__author__ = "Adam Atia"
+# Using Andrew Lee's formulation of ASM1 as a template
+
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("ADM1ReactionParameterBlock")
+class ADM1ReactionParameterData(ReactionParameterBlock):
+    """
+    Property Parameter Block Class
+    """
+
+    def build(self):
+        """
+        Callable method for Block construction.
+        """
+        super().build()
+
+        self._reaction_block_class = ADM1ReactionBlock
+
+        # Reaction Index
+        # Reaction names based on standard numbering in ADM1 paper
+        # R1:  Disintegration
+        # R2:  Hydrolysis of carbohydrates
+        # R3:  Hydrolysis of proteins
+        # R4:  Hydrolysis of lipids
+        # R5:  Uptake of sugars
+        # R6:  Uptake of amino acids
+        # R7:  Uptake of long chain fatty acids (LCFAs)
+        # R8:  Uptake of valerate
+        # R9:  Uptake of butyrate
+        # R10: Uptake of propionate
+        # R11: Uptake of acetate
+        # R12: Uptake of hydrogen
+        # R13: Decay of X_su
+        # R14: Decay of X_aa
+        # R15: Decay of X_fa
+        # R16: Decay of X_c4
+        # R17: Decay of X_pro
+        # R18: Decay of X_ac
+        # R19: Decay of X_h2
+
+        self.rate_reaction_idx = pyo.Set(initialize=[f"R{i}" for i in range(1, 20)])
+
+        # TODO: for all carbon and nitrogen content values with "varies" need to revisited; putting 0s for now
+        # Carbon content
+        Ci_dict = {
+            "S_su": 0.0313,
+            "S_aa": 0.03,  # varies
+            "S_fa": 0.0217,
+            "S_va": 0.0240,
+            "S_bu": 0.0250,
+            "S_pro": 0.0268,
+            "S_ac": 0.0313,
+            "S_h2": 0,
+            "S_ch4": 0.0156,
+            "S_IC": 1,
+            "S_IN": 0,
+            "S_I": 0.03,  # varies
+            "X_c": 0.02786,  # varies
+            "X_ch": 0.0313,
+            "X_pr": 0.03,  # varies
+            "X_li": 0.0220,
+            "X_su": 0.0313,
+            "X_aa": 0.0313,
+            "X_fa": 0.0313,
+            "X_c4": 0.0313,
+            "X_pro": 0.0313,
+            "X_ac": 0.0313,
+            "X_h2": 0.0313,
+            "X_I": 0.03,  # varies
+            # "S_an": 0,
+            # "S_cat": 0,
+        }
+        # TODO: Ci might need to go to the property model instead to be used for conc_mol_phase_comp
+        self.Ci = pyo.Var(
```

Comment:
This looks like it belongs here - the thermo properties are only concerned about actual components, not elemental concentrations (until someone want's to do elemental balances, but then you wouldn't use these reactions).

Reply:
So this ties to another issue related to the Petersen Matrix. All components are on a kg COD/m3 basis, except for inorganic carbon and inorganic nitrogen, which are in units kmol C/m3 and kmol N/m3, respectively. This `Ci` converts the mass concentration to molarity for inorganic carbon, and I've been debating with myself on the _best_ way to handle the discrepancy in units while staying true to the form taken in the literature.

Reply:
Look at the ASM1 and ASM2d example - they have similar issues with alkalinity. You will see that there is a hard coded conversion for molecular weight of C and N in those models to handle the conversion.

Reply:
I kept the conversion to keep the model (and hence parameter and stoichiometric values) in-line with the published material.